src/main/java/frc/robot/subsystems/ball/intake/IntakeImpl.java

```diff
@@ -0,0 +1,51 @@
+package frc.robot.subsystems.ball.intake;
+
+import edu.wpi.first.wpilibj.Solenoid;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import ca.team3161.lib.robot.subsystem.RepeatingPooledSubsystem;
+import java.util.concurrent.TimeUnit;
+
+public class IntakeImpl extends RepeatingPooledSubsystem implements Intake {
+
+    Solenoid intakeSolenoid;
+    WPI_TalonSRX intakeMotorController;
+    boolean extended;
+    int intakeDirection;
+    double intakeSpeed;
+
+    public IntakeImpl() {
+        super(250, TimeUnit.MILLISECONDS); // Figure out actual value
+        this.intakeSolenoid = new Solenoid(frc.robot.RobotMap.INTAKE_SOLENOID_CHANNEL);
+        this.intakeMotorController = new WPI_TalonSRX(frc.robot.RobotMap.INTAKE_TALON_PORT);
+        this.extended = this.intakeSolenoid.get();
+        this.intakeSpeed = 0.8d;
+    }
+    
+    public void extend(int motorDirection) {
+        this.intakeDirection = motorDirection;
+        this.extended = true;
+    }
+
+    public void extend() {
+        extend(1);
+    }
+
+    public void retract() {
+        this.extended = false;
+    }
+
+    public void defineResources() {
+        require(intakeMotorController);
+        require(intakeSolenoid);
+    }
+
+    public void task(){
+        if(this.extended) {
+            this.intakeSolenoid.set(true);
+            this.intakeMotorController.set(this.intakeDirection * this.intakeSpeed);
+        } else {
+            this.intakeSolenoid.set(false);
+            this.intakeMotorController.set(0.0d);
+        }
+    }
```

Comment:
Yea, it probably "feels" inefficient when you look at this since 99.99% of the time when this method is executed, it sees the same state as the last time and just sets the same values on the solenoid and controller as they already have. Luckily, those should be very cheap operations to perform, so there's very little performance penalty to doing this.

Technically, in order to be sure that the task() thread and whatever thread calls extend/retract (which could be the TitanBot thread, or a Gamepad thread, or another subsystem's thread), always agree on the value of the variable, `extended` should be declared `volatile`. Otherwise there is a possible timing window where the threads each have their own local copy of the variable which can become desynchronized from each other. This is unlikely to cause any actual issues though, so I wouldn't worry too much about this class of memory consistency bug for our purposes.